packages/react/src/components/Dropdown/DropdownItem.tsx

```diff
@@ -55,8 +63,8 @@ class DropdownItem extends UIComponent<ReactProps<DropdownItemProps>, any> {
     }),
     accessibilityItemProps: PropTypes.object,
     active: PropTypes.bool,
-    content: PropTypes.string,
-    header: PropTypes.string,
+    content: customPropTypes.itemShorthand,
```

Comment:
`content` and `header` need to become an actual shorthand so you need to use the shorthand API for this when passing them to `ListItem`. See how we do it in `ListItem.tsx` or follow the snippet below:

```jsx
<ListItem
  className={classes.root}
  header={Box.create(header, {
    defaultProps: {
      className: DropdownItem.slotClassNames.header,
      styles: styles.header,
    },
  })}
  onClick={this.handleClick}
  media={Image.create(image, {
    defaultProps: {
      className: DropdownItem.slotClassNames.image,
      avatar: true,
    },
  })}
  content={Box.create(content, {
    defaultProps: {
      className: DropdownItem.slotClassNames.content,
      styles: styles.content,
    },
  })}
  {...accessibilityItemProps}
  {...unhandledProps}
/>
```

In this way we don't need `slotClassNames` to be exported because user can target the slots directly in order to add their custom styles:

```jsx
<Provider
  theme={{
    componentStyles: {
      DropdownItem: {
        root: { /* custom styles */},
        header: { /* custom styles */},
        content: { /* custom styles */},
      },
    },
  }}
>
```

@mnajdova can you confirm?